.github/workflows/ui-tests/cypress/integration/deposit_spec.js

```diff
@@ -0,0 +1,192 @@
+        cy.get('#current-balance').then(($span) => {
+            const updatedBalanceSpan = $span.text()
+            cy.wrap(updatedBalanceSpan).should('contain', expectedBalance)
+        })
+
+    })
+
+    it('can see transaction in history after deposit', function () {
+        const depositAmount = validPayment()
+
+        cy.deposit(externalAccount, depositAmount)
+        cy.get('.alert').contains(depositMsgs.success)
+
+        cy.visit('/home')
+
+        cy.get('#transaction-table').find('tbody>tr').as('latest')
+
+        cy.get('@latest').find('.transaction-account').contains(externalAccount.accountNum)
+        cy.get('@latest').find('.transaction-type').contains('Debit')
+        cy.get('@latest').find('.transaction-amount').contains(depositAmount)
+    })
+
+    it('can deposit to new account and see new account', function () {
+        const accountNum = validAccountNum()
+        const routingNum = validRoutingNum()
+        const newExternalAccount = {
+            accountNum: accountNum,
+            routingNum: routingNum,
+            contactLabel: `testcontact${accountNum}`
+        }
+        const paymentAmount = validPayment()
+
+        cy.depositToNewAccount(newExternalAccount, paymentAmount)
+        cy.get('.alert').contains(depositMsgs.success)
+
+        cy.reload()
+        cy.get('#depositSpan').click()
+        cy.get('#accounts').contains(newExternalAccount.contactLabel)
+        cy.get('#accounts').contains(newExternalAccount.accountNum)
+        cy.get('#accounts').contains(newExternalAccount.routingNum)
+    })
+
+})
+
+describe('Deposit is unsuccessful with invalid data', function () {
+    beforeEach(function () {
+        cy.loginRequest(username, password)
+        cy.visit('/home')
+    })
+
+    it('cannot be equal to zero', function () {
+        const zeroPayment = 0
+        cy.deposit(externalAccount, zeroPayment)
+        cy.get('.invalid-feedback').should('be.visible')
+        cy.get('.invalid-feedback').contains(invalidFeedback.payment)
+    })
+
+    it('cannot be less than zero', function () {
+        const negativePayment = `-${validPayment()}`
+        cy.deposit(externalAccount, negativePayment)
+        cy.get('.invalid-feedback').should('be.visible')
+        cy.get('.invalid-feedback').contains(invalidFeedback.payment)
+    })
+
+    it('cannot reference invalid account number', function () {
+        const invalidExternalAccount = {
+            accountNum: randomInt(100,100000),
+            routingNum: validRoutingNum(),
+            contactLabel: `testcontact invalid ${this.accountNum}`
+        }
+
+        const paymentAmount = validPayment()
+
+        cy.depositToNewAccount(invalidExternalAccount, paymentAmount)
+        cy.get('.invalid-feedback').should('be.visible')
+        cy.get('.invalid-feedback').contains(invalidFeedback.accountNum)
+    })
+
+    it('cannot reference invalid routing number', function () {
+        const invalidExternalAccount = {
```

Comment:
it might be interesting to loop through a bunch of invalid values

Reply:
I think it would be best to hold off on that until screenshots are enabled because of how cypress errors are reported in log